src/java/us/kbase/templates/python_server.vm.properties

```diff
@@ -129,11 +129,10 @@ class JSONRPCServiceCustom(JSONRPCService):
             # Exception was raised inside the method.
             newerr = JSONServerError()
             newerr.trace = traceback.format_exc()
-            if isinstance(e.message, str):
-                newerr.data = e.message
+            if len(e.args) == 1:
+                newerr.data = str(e.args[0])
             else:
-                # Some exceptions embed other exceptions as the message
-                newerr.data = repr(e.message)
+                newerr.data = str(e.args)
```

Comment:
I specifically remember switching from str to repr because there was info that was being lost with str. I don't remember the specifics though, that was years ago.

Reply:
Based on the comment, it was for embedded exceptions. I think all that info still would be present in the traceback though

Reply:
I think it was actually the error name but you can rest easy. `args` does not have the same issue as message.

```>>> try:
...     func()
... except NameError as e:
...     meh = ValueError(e)
... 
>>> meh.message
NameError("name 'func' is not defined",)
>>> str(meh.message)
"name 'func' is not defined"
>>> repr(meh.message)
'NameError("name \'func\' is not defined",)'
>>> str(meh.args)
'(NameError("name \'func\' is not defined",),)'
>>> repr(meh.args)
'(NameError("name \'func\' is not defined",),)'```

Reply:
`args[0]` does have that issue though:

```python
In [8]: def func():
    def inner():
        raise IOError('io')
    try:
        inner()
    except IOError as e:
        raise ValueError(e)
   ...:     

In [9]: try:     
    func()                  
except ValueError as e:
    err = e
   ...:     

In [10]: str(err)
Out[10]: 'io'

In [11]: repr(err)
Out[11]: "ValueError(IOError('io',),)"

In [12]: str(err.args[0])
Out[12]: 'io'

In [13]: repr(err.args[0])
Out[13]: "IOError('io',)"

In [14]: str(err.args)
Out[14]: "(IOError('io',),)"

In [15]: repr(err.args)
Out[15]: "(IOError('io',),)"
```